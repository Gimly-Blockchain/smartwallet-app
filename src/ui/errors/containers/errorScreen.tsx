import React from 'react'
import { StyleSheet, Text, View } from 'react-native'
import { JolocomButton, Wrapper } from '../../structure'
import { NavigationScreenProp, NavigationState } from 'react-navigation'
import { centeredText, fontMain } from '../../../styles/typography'
import { Colors } from '../../../styles'
import { BP } from '../../../styles/breakpoints'
import { BlueCircleImage, RedCircleImage } from '../../../resources'

const styles = StyleSheet.create({
  wrapper: {
    backgroundColor: Colors.overflowBlack,
    justifyContent: 'space-between',
  },
  imageWrapper: {
    marginTop: '15%',
    width: '65%',
    height: '54%',
  },
  title: {
    paddingHorizontal: 20,
    fontFamily: fontMain,
    fontSize: BP({
      small: 26,
      medium: 28,
      large: 28,
    }),
    color: Colors.sandLight090,
    ...centeredText,
  },
  message: {
    marginTop: 13,
    paddingHorizontal: 20,
    fontFamily: fontMain,
    fontSize: BP({
      small: 18,
      medium: 20,
      large: 20,
    }),
    color: Colors.sandLight080,
    lineHeight: 24,
    ...centeredText,
  },
  interactButton: {
    marginTop: 14,
    borderColor: 'white',
    borderWidth: 0.5,
  },
  buttonSection: {
    width: '100%',
    paddingBottom: 62,
    paddingHorizontal: 20,
    justifyContent: 'space-between',
  },
  contentSection: {
    width: '100%',
    alignItems: 'center',
  },
})

enum ImageType {
  Red = 'red',
  Blue = 'blue',
}

export interface ErrorScreenParams {
  title: string
  message: string
  image?: ImageType
  interact?: {
    label: string
    onInteract: () => void | Promise<void>
  }
  dismiss: {
    label: string
    onDismiss: () => void | Promise<void>
  }
}

interface Props {
  navigation: NavigationScreenProp<NavigationState, ErrorScreenParams>
}

export const ErrorScreenContainer: React.FC<Props> = props => {
  const {
    navigation: {
      state: {
        params: { title, message, interact, dismiss, image },
      },
    },
  } = props

  return (
    <Wrapper style={styles.wrapper}>
      <View style={styles.contentSection}>
        <View style={styles.imageWrapper}>
          {image === ImageType.Red ? <RedCircleImage /> : <BlueCircleImage />}
        </View>
        <Text style={styles.title}>{title}</Text>
        <Text style={styles.message}>{message}</Text>
      </View>
      <View style={styles.buttonSection}>
        <JolocomButton
          onPress={dismiss.onDismiss}
          transparent
          text={dismiss.label}
        />
        {interact && (
          <JolocomButton
            containerStyle={styles.interactButton}
            transparent
            onPress={interact.onInteract}
            text={interact.label}
          />
        )}
      </View>
    </Wrapper>
  )
}
