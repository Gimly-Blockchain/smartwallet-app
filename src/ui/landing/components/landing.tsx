import React, { ReactNode } from 'react'
import { StyleSheet, Dimensions } from 'react-native'
import { Button } from 'react-native-material-ui'
import { Block, Container, CenteredText } from 'src/ui/structure'
import { JolocomTheme } from 'src/styles/jolocom-theme'
import I18n from 'src/locales/i18n'
import { Landing00, Landing01, Landing02, Landing03 } from 'src/resources'
import strings from 'src/locales/strings'

const Carousel = require('react-native-snap-carousel').default
const Pagination = require('react-native-snap-carousel').Pagination

interface State {
  activeSlide: number
}

interface Props {
  handleGetStarted: () => void
  handleRecover: () => void
}

interface Slide {
  svgImage: ReactNode
  title: string
  infoText: string
}

const viewWidth: number = Dimensions.get('window').width
const headerFontSize =
  viewWidth < 360
    ? JolocomTheme.landingHeaderFontSizeSmall
    : JolocomTheme.landingHeaderFontSize
const labelFontSize =
  viewWidth < 360 ? JolocomTheme.labelFontSizeSmall : JolocomTheme.labelFontSize

const styles = StyleSheet.create({
  mainContainerStyle: {
    paddingTop: 0,
    backgroundColor: '#05050d',
    justifyContent: 'flex-end',
    flexDirection: 'column',
    flex: 1,
  },
  carouselTextContainer: {
    paddingHorizontal: viewWidth / 18,
    flex: 0.4,
    marginTop: 'auto',
    justifyContent: 'flex-end',
    backgroundColor: 'transparent',
  },
  activeDotStyle: {
    width: 8,
    height: 8,
    backgroundColor: JolocomTheme.dotColorActive,
  },
  inactiveDotStyle: {
    width: 4,
    height: 4,
    opacity: 0.6,
    backgroundColor: JolocomTheme.dotColorInactive,
  },
  header: {
    color: JolocomTheme.primaryColorSand,
    fontFamily: JolocomTheme.contentFontFamily,
    fontSize: headerFontSize,
    fontWeight: '100',
  },
  subHeader: {
    color: JolocomTheme.primaryColorSand,
    opacity: 0.8,
    fontFamily: JolocomTheme.contentFontFamily,
    fontSize: labelFontSize,
    fontWeight: '100',
    lineHeight: labelFontSize + 4,
    marginTop: 15,
  },
  paginationBlock: {
    flex: 0.15,
    backgroundColor: '#05050d',
  },
  buttonBlock: {
    flex: 0.3,
    backgroundColor: '#05050d',
  },
  buttonContainer: {
    marginTop: 5,
    height: 48,
    minWidth: 164,
    borderRadius: 4,
    backgroundColor: JolocomTheme.primaryColorPurple,
  },
  buttonText: {
    paddingVertical: 15,
    fontFamily: JolocomTheme.contentFontFamily,
    color: JolocomTheme.primaryColorWhite,
    fontSize: JolocomTheme.headerFontSize,
    fontWeight: '100',
    textAlign: 'center',
    minWidth: 158,
  },
})

const carouselInfo: Slide[] = [
  {
    svgImage: <Landing00 />,
    title: I18n.t(strings.YOUR_JOLOCOM_WALLET),
    infoText:
      I18n.t(
        strings.TAKE_BACK_CONTROL_OF_YOUR_DIGITAL_SELF_AND_PROTECT_YOUR_PRIVATE_DATA_AGAINST_UNFAIR_USAGE,
      ) + '.',
  },
  {
    svgImage: <Landing01 height={'100%'} width={'100%'} />,
    title: I18n.t(strings.ITS_EASY),
    infoText:
      I18n.t(strings.FORGET_ABOUT_LONG_FORMS_AND_REGISTRATIONS) +
      '. ' +
      I18n.t(
        strings.INSTANTLY_ACCESS_SERVICES_WITHOUT_USING_YOUR_SOCIAL_MEDIA_PROFILES,
      ) +
      '.',
  },
  {
    svgImage: <Landing03 height={'100%'} width={'100%'} />,
    title: I18n.t(strings.ENHANCED_PRIVACY),
    infoText:
      I18n.t(strings.SHARE_ONLY_THE_INFORMATION_A_SERVICE_REALLY_NEEDS) +
      '. ' +
      I18n.t(strings.PROTECT_YOUR_DIGITAL_SELF_AGAINST_FRAUD) +
      '.',
  },
  {
    svgImage: <Landing02 height={'100%'} width={'100%'} />,
    title: I18n.t(strings.GREATER_CONTROL),
    infoText:
      I18n.t(
        strings.KEEP_ALL_YOUR_DATA_WITH_YOU_IN_ONE_PLACE_AVAILABLE_AT_ANY_TIME,
      ) +
      '. ' +
      I18n.t(strings.TRACK_WHERE_YOU_SIGN_IN_TO_SERVICES) +
      '.',
  },
]

export class LandingComponent extends React.Component<Props, State> {
  state = {
    activeSlide: 0,
  }

  private renderItem = ({ item }: { item: Slide }) => {
    const { svgImage, title, infoText } = item
    return (
      <Block>
        {svgImage}
        <Block style={styles.carouselTextContainer}>
          <CenteredText style={styles.header} msg={title} />
          <CenteredText style={styles.subHeader} msg={infoText} />
        </Block>
      </Block>
    )
  }

  private renderPagination() {
    const { activeSlide } = this.state
    return (
      <Pagination
        dotsLength={carouselInfo.length}
        activeDotIndex={activeSlide}
        dotStyle={styles.activeDotStyle}
        inactiveDotStyle={styles.inactiveDotStyle}
        inactiveDotOpacity={0.4}
        inactiveDotScale={0.6}
      />
    )
  }

  render() {
    return (
      <Container style={styles.mainContainerStyle}>
        <Block>
          <Carousel
            data={carouselInfo}
            renderItem={this.renderItem}
            lockScrollWhileSnapping
            lockScrollTimeoutDuration={1000}
            loop
            sliderWidth={viewWidth}
            itemWidth={viewWidth}
            layout={'default'}
            onSnapToItem={(index: number) =>
              this.setState({ activeSlide: index })
            }
          />
        </Block>
        <Block style={styles.paginationBlock}>{this.renderPagination()}</Block>
        <Block style={styles.buttonBlock}>
          <Button
            raised
            onPress={this.props.handleGetStarted}
            style={{
              container: styles.buttonContainer,
              text: styles.buttonText,
            }}
            upperCase={false}
            text={I18n.t(strings.GET_STARTED)}
          />
          <Button
            raised
            onPress={this.props.handleRecover}
            style={{
              container: styles.buttonContainer,
              text: styles.buttonText,
            }}
            upperCase={false}
            text={'Recover Identity'}
          />
        </Block>
      </Container>
    )
  }
}
