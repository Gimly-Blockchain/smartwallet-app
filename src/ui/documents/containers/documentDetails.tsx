import React from 'react'
import { connect } from 'react-redux'
import { NavigationScreenProps } from 'react-navigation'
import { View, StyleSheet, ScrollView } from 'react-native'
import { ClaimInterface } from 'cred-types-jolocom-core'
import { RootState } from 'src/reducers/'
import I18n from 'src/locales/i18n'
import { DocumentCard } from '../components/documentCard'
import { DocumentDetails as DocumentDetailsComponent } from '../components/documentDetails'
import strings from '../../../locales/strings'
import { JolocomTheme } from 'src/styles/jolocom-theme.ios'

interface Props
  extends ReturnType<typeof mapStateToProps>,
    NavigationScreenProps {}

interface State {}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: JolocomTheme.primaryColorGrey,
  },
  documentCardContainer: {
    paddingVertical: 15,
    justifyContent: 'center',
    alignItems: 'center',
  },
})

export class DocumentDetailsContainer extends React.Component<Props, State> {
  public static navigationOptions = ({
    navigation,
    navigationOptions,
  }: NavigationScreenProps) => ({
    ...navigationOptions,
    // FIXME needs to be changed to this on update to react-navigation
    // headerTitle: navigation.getParam('headerTitle', ''),
    headerTitle: navigation.state.params && navigation.state.params.headerTitle,
  })

  public componentWillMount() {
    const { document } = this.props
    // FIXME use filter from src/lib/filterDecoratedClaims
    const isExpired = document.expires
      ? document.expires.valueOf() < new Date().valueOf()
      : true

    let headerTitle = (document.claimData as ClaimInterface).type
    if (isExpired) headerTitle = `[${I18n.t(strings.EXPIRED)}] ${headerTitle}`
    this.props.navigation.setParams({ headerTitle })
  }

  public render(): JSX.Element {
    const { document } = this.props

    return (
      <View style={styles.container}>
        <View style={styles.documentCardContainer}>
          <DocumentCard document={document} />
        </View>
        <ScrollView>
          <DocumentDetailsComponent document={document} />
        </ScrollView>
      </View>
    )
  }
}

const mapStateToProps = (state: RootState) => ({
  document: state.documents.selectedDocument,
})

export const DocumentDetails = connect(mapStateToProps)(
  DocumentDetailsContainer,
)
