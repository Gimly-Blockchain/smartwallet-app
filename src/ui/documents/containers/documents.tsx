import React from 'react'
import { connect } from 'react-redux'
import { Animated, ScrollView, StyleSheet, Text, View } from 'react-native'

import { openDocumentDetails } from 'src/actions/documents'
import { DecoratedClaims } from 'src/reducers/account'
import { RootState } from 'src/reducers'
import { getDocumentClaims } from 'src/utils/filterDocuments'
import { ThunkDispatch } from 'src/store'

import I18n from 'src/locales/i18n'
import { filters } from 'src/lib/filterDecoratedClaims'

import { DocumentsCarousel } from '../components/documentsCarousel'
import { DocumentsList } from '../components/documentsList'
import { DocumentViewToggle } from '../components/documentViewToggle'
import strings from '../../../locales/strings'
import { Typography, Colors, Spacing } from 'src/styles'

interface Props
  extends ReturnType<typeof mapDispatchToProps>,
    ReturnType<typeof mapStateToProps> {}

interface State {
  activeDocumentIndex: number
  showingValid: boolean
}

/*
const APPBAR_HEIGHT = Platform.select({
  ios: 44,
  android: 56,
  default: 64,
})
*/

const styles = StyleSheet.create({
  mainContainer: {
    flex: 1,
    backgroundColor: Colors.lightGreyLighter,
  },
  topContainer: {
    justifyContent: 'center',
    alignItems: 'center',
    paddingVertical: Spacing.MD,
  },
  emptyDocumentsContainer: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
    paddingHorizontal: Spacing.MD,
  },
  emptyDocumentsText: {
    ...Typography.mainText,
    ...Typography.centeredText,
    color: Colors.greyLight,
  },
})

export class DocumentsContainer extends React.Component<Props, State> {
  public ScrollViewRef: ScrollView | null = null

  public state = {
    activeDocumentIndex: 0,
    showingValid: true,
  }

  private scrollToTop(): void {
    if (this.ScrollViewRef) {
      this.ScrollViewRef.scrollTo({ x: 0, y: 0, animated: true })
    }
  }

  private onActiveIndexChange(index: number): void {
    this.setState({ activeDocumentIndex: index })
    this.scrollToTop()
  }

  private handleToggle = () => {
    this.setState((prevState: State) => ({
      showingValid: !prevState.showingValid,
      activeDocumentIndex: 0,
    }))
    this.scrollToTop()
  }

  public render(): JSX.Element {
    const { openDocumentDetails, decoratedCredentials } = this.props
    const documents = getDocumentClaims(decoratedCredentials['Other'])
    const docFilter = this.state.showingValid
      ? filters.filterByValid
      : filters.filterByExpired
    const displayedDocs = docFilter(documents)
    const isEmpty = displayedDocs.length === 0
    const otherIsEmpty = displayedDocs.length === documents.length

    return (
      <Animated.View style={styles.mainContainer}>
        {!otherIsEmpty && (
          <DocumentViewToggle
            showingValid={this.state.showingValid}
            onTouch={this.handleToggle}
          />
        )}
        <ScrollView
          // scrollEventThrottle={16}
          // onScroll={Animated.event([
          //   {
          //     nativeEvent: {
          //       contentOffset: { y: this.state.headerHeight },
          //     },
          //   },
          // ])}
          // to scroll to top upon changing card
          ref={ref => (this.ScrollViewRef = ref)}
          contentContainerStyle={isEmpty && { flex: 1 }}
          scrollEnabled={!isEmpty}
        >
          {isEmpty ? (
            <View style={styles.emptyDocumentsContainer}>
              <Text style={styles.emptyDocumentsText}>
                {I18n.t(strings.NO_DOCUMENTS_TO_SEE_HERE) + '...'}
              </Text>
            </View>
          ) : this.state.showingValid ? (
            <DocumentsCarousel
              documents={displayedDocs}
              activeIndex={this.state.activeDocumentIndex}
              onActiveIndexChange={this.onActiveIndexChange.bind(this)}
            />
          ) : (
            <DocumentsList
              documents={displayedDocs}
              onDocumentPress={openDocumentDetails}
            />
          )}
        </ScrollView>
      </Animated.View>
    )
  }
}

const mapStateToProps = ({
  account: {
    claims: { decoratedCredentials },
  },
}: RootState) => ({
  decoratedCredentials,
})

const mapDispatchToProps = (dispatch: ThunkDispatch) => ({
  openDocumentDetails: (doc: DecoratedClaims) =>
    dispatch(openDocumentDetails(doc)),
})

export const Documents = connect(
  mapStateToProps,
  mapDispatchToProps,
)(DocumentsContainer)
