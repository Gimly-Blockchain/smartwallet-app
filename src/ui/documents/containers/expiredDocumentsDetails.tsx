import React from 'react'
import { ExpiredDocumentsDetailsComponent } from 'src/ui/documents/components/expiredDocumentDetails'
import { connect } from 'react-redux'
import { RootState } from 'src/reducers/'
import { NavigationScreenProps } from 'react-navigation'
import { DecoratedClaims } from 'src/reducers/account'

interface ConnectProps {
  selectedExpiredDocument: DecoratedClaims
}

interface Props extends ConnectProps {
  documents: Document[]
}

interface State {}

export class ExpiredDocumentsDetailsContainer extends React.Component<
  Props,
  State
> {
  public static navigationOptions = ({
    navigation,
    navigationOptions,
  }: NavigationScreenProps) => ({
    ...navigationOptions,
    headerTitle: navigation.getParam('headerTitle', 'DEFAULT'),
  })

  public render(): JSX.Element {
    const { selectedExpiredDocument } = this.props
    return (
      <ExpiredDocumentsDetailsComponent document={selectedExpiredDocument} />
    )
  }
}

const mapStateToProps = (state: RootState): {} => ({
  selectedExpiredDocument: state.documents.selectedExpiredDocument,
})

const mapDispatchToProps = (dispatch: Function): {} => ({})

export const ExpiredDocumentsDetails = connect(
  mapStateToProps,
  mapDispatchToProps,
)(ExpiredDocumentsDetailsContainer)
