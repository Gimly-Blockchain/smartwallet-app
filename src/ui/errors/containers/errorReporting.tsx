import React, { useState } from 'react'
import { connect } from 'react-redux'
import { NavigationScreenProp, NavigationState } from 'react-navigation'
import { AppError, reportError } from '../../../lib/errors'
import { ThunkDispatch } from '../../../store'
import { routeList } from '../../../routeList'
import { navigationActions } from '../../../actions'
import { Emoji, EmojiSection } from '../components/emojiSection'
import { Container } from '../../structure'
import { styles } from '../styles'
import { ScrollView } from 'react-native'
import { ChooseIssueSection } from '../components/chooseIssueSection'
import { DescriptionSection } from '../components/descriptionSection'
import { ContactSection } from '../components/contactSection'
import { GradientButton } from '../../structure/gradientButton'
import I18n from '../../../locales/i18n'
import strings from '../../../locales/strings'
interface PaymentNavigationParams {
  error: AppError | Error | undefined
}

interface Props extends ReturnType<typeof mapDispatchToProps> {
  navigation: NavigationScreenProp<NavigationState, PaymentNavigationParams>
}

export enum Inputs {
  None,
  Dropdown,
  Description,
  Contact,
}

const ErrorReportingContainer = (props: Props) => {
  const { navigateToScreen, navigation } = props

  const [pickedIssue, setIssue] = useState<string>()
  const [description, setDescription] = useState<string>('')
  const [contact, setContact] = useState<string>('')
  const [currentInput, setInput] = useState<Inputs>(Inputs.None)
  const [toggleState, setToggle] = useState<boolean>(false)
  const [selectedEmoji, setEmoji] = useState<Emoji>(Emoji.Empty)

  const onSubmitReport = () => {
    const { error } = props.navigation.state.params

    const userReport = {
      userError: pickedIssue,
      userDescription: description,
      userContact: contact,
    }
    console.log(userReport)

    if (navigation && error) {
      reportError({ ...userReport, error })
      if (error instanceof AppError) {
        navigateToScreen(error.navigateTo)
      } else {
        navigateToScreen(routeList.AppInit)
      }
    }
  }

  return (
    <Container style={styles.wrapper}>
      <ScrollView>
        <ChooseIssueSection
          currentInput={currentInput}
          pickedIssue={pickedIssue}
          onIssuePick={setIssue}
          setInput={setInput}
        />
        <DescriptionSection
          currentInput={currentInput}
          setInput={setInput}
          setDescription={setDescription}
          toggleState={toggleState}
          setToggle={setToggle}
          description={description}
        />
        <ContactSection
          onContactInput={setContact}
          currentInput={currentInput}
          setInput={setInput}
          contactValue={contact}
        />
        <EmojiSection selectedEmoji={selectedEmoji} setEmoji={setEmoji} />
        <GradientButton
          onPress={onSubmitReport}
          containerStyle={{
            marginTop: 45,
            marginBottom: 66,
          }}
          text={I18n.t(strings.SUBMIT_REPORT)}
        />
      </ScrollView>
    </Container>
  )
}

const mapDispatchToProps = (dispatch: ThunkDispatch) => ({
  navigateToScreen: (screen: routeList) =>
    dispatch(navigationActions.navigate({ routeName: screen })),
})

export const ErrorReporting = connect(
  null,
  mapDispatchToProps,
)(ErrorReportingContainer)
