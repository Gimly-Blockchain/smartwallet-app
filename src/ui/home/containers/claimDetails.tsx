import React from 'react'
import { ClaimDetailsComponent } from 'src/ui/home/components/claimDetails'
import { connect } from 'react-redux'
import { RootState } from 'src/reducers/'
import { accountActions } from 'src/actions'
import { ThunkDispatch } from '../../../store'
import { withErrorHandling, withLoading } from '../../../actions/modifiers'
import { showErrorScreen } from '../../../actions/generic'
import { setDeepLinkLoading } from '../../../actions/sso'

interface Props
  extends ReturnType<typeof mapDispatchToProps>,
    ReturnType<typeof mapStateToProps> {}

interface State {}

export class ClaimDetailsContainer extends React.Component<Props, State> {
  render() {
    return (
      <ClaimDetailsComponent
        saveClaim={this.props.saveClaim}
        handleClaimInput={this.props.handleClaimInput}
        selectedClaim={this.props.claims.selected}
      />
    )
  }
}

const mapStateToProps = ({ account: { claims } }: RootState) => ({
  claims,
})

const mapDispatchToProps = (dispatch: ThunkDispatch) => ({
  saveClaim: () =>
    dispatch(
      withLoading(setDeepLinkLoading)(
        withErrorHandling(showErrorScreen)(accountActions.saveClaim),
      ),
    ),
  handleClaimInput: (fieldValue: string, fieldName: string) => {
    dispatch(accountActions.handleClaimInput(fieldValue, fieldName))
  },
})

export const ClaimDetails = connect(
  mapStateToProps,
  mapDispatchToProps,
)(ClaimDetailsContainer)
