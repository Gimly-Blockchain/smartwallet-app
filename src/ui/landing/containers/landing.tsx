import React from 'react'
import { connect } from 'react-redux'
import { LandingComponent } from 'src/ui/landing/components/landing'
import { registrationActions } from 'src/actions/'
import { withErrorHandling } from '../../../actions/modifiers'
import { showErrorScreen } from '../../../actions/generic'
import { AppError, ErrorCode } from '../../../lib/errors'
import { routeList } from '../../../routeList'
import { ThunkDispatch } from '../../../store'
import { StatusBar } from 'react-native'

interface Props extends ReturnType<typeof mapDispatchToProps> {}

export class LandingContainer extends React.Component<Props> {
  render() {
    return (
      <React.Fragment>
        <StatusBar barStyle="light-content" />
        <LandingComponent handleButtonTap={this.props.startRegistration} />
      </React.Fragment>
    )
  }
}

const mapDispatchToProps = (dispatch: ThunkDispatch) => ({
  startRegistration: () =>
    dispatch(
      withErrorHandling(
        showErrorScreen,
        (err: AppError) =>
          new AppError(ErrorCode.RegistrationFailed, err, routeList.Landing),
      )(registrationActions.startRegistration),
    ),
})

export const Landing = connect(
  null,
  mapDispatchToProps,
)(LandingContainer)
