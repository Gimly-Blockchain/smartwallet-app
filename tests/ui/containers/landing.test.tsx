import React from 'react'
import { LandingContainer } from 'src/ui/landing/containers/landing'
import { shallow } from 'enzyme'

describe('landing container', () => {
  it('mounts correctly and matches snapshot', () => {
    const props: LandingContainer['props'] = {
      openInitAction: jest.fn(),
    }

    const rendered = shallow(<LandingContainer {...props} />)
    expect(rendered).toMatchSnapshot()
    expect(props.openInitAction).not.toHaveBeenCalled()
  })
})
