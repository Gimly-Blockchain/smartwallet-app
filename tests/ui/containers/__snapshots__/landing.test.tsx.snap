// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`landing container mounts correctly and matches snapshot 1`] = `
<Fragment>
  <StatusBar
    animated={false}
    barStyle="light-content"
    showHideTransition="fade"
  />
  <LandingComponent
    handleGetStarted={[MockFunction]}
    handleRecover={[MockFunction]}
  />
</Fragment>
`;
